cors.go

```diff
@@ -1 +1,132 @@
+// Copyright 2021 Flamego. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+// Package cors is a middleware that generates CORS headers for Flamego.
 package cors
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/flamego/flamego"
+)
+
+const anyDomain = "!*"
+
+// Options to configure the CORS middleware read from the [cors] section of the ini configuration file.
+// ref: https://stackoverflow.com/questions/54300997/is-it-possible-to-cache-http-options-response?noredirect=1#comment95790277_54300997
+// ref: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Max-Age
+type Options struct {
+	// SCHEME may be http or https as accepted schemes or the '*' wildcard to accept any scheme.
+	Scheme string
+	// ALLOW_DOMAIN may be a comma separated list of domains that are allowed to run CORS requests
+	// Special values are the  a single '*' wildcard that will allow any domain to send requests without
+	// credentials and the special '!*' wildcard which will reply with requesting domain in the 'access-control-allow-origin'
+	// header and hence allow requests from any domain *with* credentials.
+	AllowDomain []string
+	// AllowSubdomain allowed
+	AllowSubdomain bool
+	// METHODS may be a comma separated list of HTTP-methods to be accepted.
+	Methods []string
+	// MAX_AGE_SECONDS may be the duration in secs for which the response is cached (default 600).
+	MaxAgeSeconds int
+	// ALLOW_CREDENTIALS set to false rejects any request with credentials.
+	AllowCredentials bool
```

Comment:
```suggestion
	// Scheme may be http or https as accepted schemes or the '*' wildcard to accept any scheme.
	Scheme string
	// AllowDomains is a comma separated list of domains that are allowed to run CORS requests
	// Special values are the  a single '*' wildcard that will allow any domain to send requests without
	// credentials and the special '!*' wildcard which will reply with requesting domain in the 'access-control-allow-origin'
	// header and hence allow requests from any domain *with* credentials.
	AllowDomains []string
	// AllowSubdomain allowed ???
	AllowSubdomain bool
	// Methods is a comma separated list of HTTP-methods to be accepted.
	Methods []string
	// MaxAgeSeconds is the duration in secs for which the response is cached (default 600).
	MaxAgeSeconds int
	// AllowCredentials set to false rejects any request with credentials.
	AllowCredentials bool
```

1. Please finish the incomplete sentence.
2. Please describe what are the default values for each field if left unset.